Bonsai.Miniscope/UCLAMiniscope.cs

```diff
@@ -50,30 +52,29 @@ public enum FPS
         // State
         IObservable<IplImage> source;
         readonly object captureLock = new object();
-        //readonly CapturePropertyCollection captureProperties = new CapturePropertyCollection();
 
         // Functor
         public UCLAMiniscope()
         {
-            lastLEDBrightness = LEDBrightness;
-            lastExposure = Exposure;
-            lastSensorGain = SensorGain;
-
             source = Observable.Create<IplImage>((observer, cancellationToken) =>
             {
                 return Task.Factory.StartNew(() =>
                 {
                     lock (captureLock)
                     {
+                        var lastRecordingFramePulse = false;
```

Comment:
Yes, by having the variables declared at class-level, they will be shared by all subscriptions of the observable stream.

Tbh, I don't think this is ever really a problem in this case, since OpenCV will struggle if you try to access the same camera index twice anyway, but it's good practice when developing reactive operators to keep variables as local as they can be.